File: 81-search-in-rotated-sorted-array-ii/81-search-in-rotated-sorted-array-ii.java
class Solution {
    
    public static boolean binarySearch(int left, int right, int target,int[] nums){
        while (left <= right){ 
        int mid = left + (right - left)/2;
        if(nums[mid]==target) return true;
        else if (nums[mid] < target) left = mid + 1;
        else right = mid - 1;
        }
        return false;    
    }
    
    public boolean search(int[] nums, int target) {
        int index = 0;
        for(int i=1; i<nums.length;i++){
            if(nums[i-1] > nums[i]){
                index = i;
                break;}
        }
        return binarySearch(0, index-1,target,nums) || binarySearch(index, nums.length-1,target,nums);
        
    }
    
}

File: 81-search-in-rotated-sorted-array-ii/81-search-in-rotated-sorted-array-ii.py
class Solution:
    def search(self, nums: List[int], target: int) -> bool:
        index = 0
        for i in range(1,len(nums)):
            if nums[i-1] > nums[i]:
                index = i
                break
        
        def binarySearch(left, right):
            while left <= right:
                mid = left + (right - left)//2

                if nums[mid] == target:
                    return True
                elif nums[mid] < target:
                    left = mid + 1
                else:
                    right = mid - 1

            return False
        
        return binarySearch(0, index-1) or binarySearch(index,len(nums)-1)





















        # nums.sort()
#         left = 0
#         right = len(nums) - 1
        
#         while left <= right:
#             mid = left + (right - left)//2
            
#             if nums[mid] == target:
#                 return True
#             elif nums[mid] < target:
#                 left = mid + 1
#             else:
#                 right = mid - 1
            
#         return False